controllers/applicationvolumereplication_controller.go

```diff
@@ -0,0 +1,382 @@
+/*
+Copyright 2021 The RamenDR authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	spokeClusterV1 "github.com/open-cluster-management/api/cluster/v1"
+	ocmworkv1 "github.com/open-cluster-management/api/work/v1"
+	plrv1 "github.com/open-cluster-management/multicloud-operators-placementrule/pkg/apis/apps/v1"
+	"github.com/open-cluster-management/multicloud-operators-placementrule/pkg/utils"
+	subv1 "github.com/open-cluster-management/multicloud-operators-subscription/pkg/apis/apps/v1"
+
+	"github.com/go-logr/logr"
+	errorswrapper "github.com/pkg/errors"
```

Comment:
Why rename errors to errorswrapper?

Reply:
The packages `github.com/pkg/errors` and `k8s.io/apimachinery/pkg/api/errors` are named `errors`, we need to import them differently to use the same.

`github.com/pkg/errors` helps with error wrapping, hence the name.